crmprtd/align.py

```diff
@@ -40,6 +41,29 @@
     ureg.define(def_)
 
 
+def cached_function(attrs):
+    """ """
+
+    def wrapper(f):
+        cache = {}
+
+        def memoize(sesh, *args, **kwargs):
+            nonlocal cache
+            key = (args) + tuple(kwargs.items())
+            if key not in cache:
+                obj = f(sesh, *args, **kwargs)
+                if not obj:
+                    return None
+                cache[key] = SimpleNamespace(
+                    **{attr: getattr(obj, attr) for attr in attrs}
+                )
+            return cache[key]
+
+        return memoize
+
+    return wrapper
```

Comment:
Cool. I assume this is to support caching on only selected args/kwargs, unlike, say, `lrucache`, which caches on all args.

Also it looks like you use `SimpleNamespace` to imitate the behaviour of a SQLAlchemy query result (row). So maybe change the function name to `cached_query`?

And add a docstring describing this?

Reply:
I'll remember `SimpleNamespace`, very handy.

Reply:
Yeah, that would have been easier to write the docstring back when I remembered how this worked! :stuck_out_tongue: